### PR TITLE
introduced fga package

### PR DESCRIPTION
**Pull request by vertex451**
I used mockery to generate a mocks for `openfgav1.OpenFGAServiceClient` interface to mock the responses. 
It asserts the input for each function's call and verifies if functions from that interface are called as many times as expected.

I've moved code without major changes, since there are clients that might already use it.
Just several alterations that I've done:
1. I moved grpc connection initiation outside the client constructor to make this code testable ([pointer](https://github.com/openmfp/golang-commons/pull/51/files#r1664385352)). Now we deal with the interface which can be mocked.
As a result I get rid of conn field and Dispose method, now this should be taken care on the package consumer side.
Old code to compare is [here](https://github.tools.sap/dxp/golang-dxp/blob/main/fga/client/client.go#L35-L54)

2. I moved interface declaration from fga/client/client.go to fga/interface.go. It will be the friendly entry point for library users.

3. Also, I moved fga/helpers.go to fga/helpers/helpers.go. I don't mind return it back, but I recon, this is better place for auxiliary functions.

### CODE REVIEW

File: fga/client/client.go [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+)
+
+var _ fga.OpenFGAClientServicer = (*OpenFGAClient)(nil)
+
+type OpenFGAClient struct {
+	client openfgav1.OpenFGAServiceClient
+	cache  *ttlcache.Cache[string, string]
+}
+
+func NewOpenFGAClient(openFGAServiceClient openfgav1.OpenFGAServiceClient) (*OpenFGAClient, error) {
```
Comment (vertex451) [inline fga/client/client.go:19]: I moved grpc connection initiation outside the constructor to make this code testable. Now we deal with the interface which can be mocked.
Old code to compare is [here](https://github.tools.sap/dxp/golang-dxp/blob/main/fga/client/client.go#L35-L54)

**Review by aaronschweig — COMMENTED** [review]
Very nice - looks good to me! One minor improvement I would see is to invoke the `mockery` command with the `--with-expecter` flag. This could make the `.On` calls a bit cleaner and more typesafe. You can see an example on how to use it here: https://github.com/openmfp/account-operator/blob/0b3e5cd2b24189972ccec52a5a62e4c9ab91059f/pkg/subroutines/extensions_test.go#L395

**Comment by vertex451** [comment]
> Very nice - looks good to me! One minor improvement I would see is to invoke the `mockery` command with the `--with-expecter` flag. This could make the `.On` calls a bit cleaner and more typesafe. You can see an example on how to use it here: https://github.com/openmfp/account-operator/blob/0b3e5cd2b24189972ccec52a5a62e4c9ab91059f/pkg/subroutines/extensions_test.go#L395

Good suggestion, done!

**Review by aaronschweig — APPROVED** [review]
🚀

**Pull request merged**